toolbox/util/githubClient.go

```diff
@@ -214,6 +214,21 @@ func (g GithubClient) ExistBranch(repo, branch string) (bool, error) {
 	return false, nil
 }
 
+// ExistTag checks if a given tag has already existed on remote repo
+func (g GithubClient) ExistTag(repo, tag string) (bool, error) {
+	tags, _, err := g.client.Repositories.ListTags(
```

Comment:
Have you looked at https://developer.github.com/v3/git/tags/#get-a-tag

Reply:
Yes the problem is that if the tag does not exist then the go github client returns an error, which is hard to distinguish from other rpc errors like unauthorized, timeout, etc.